### ISSUE TITLE
Time line syncing is broken

### ISSUE THREAD

**Issue by evgmik** [sahib/brig#56]
This bug is on a `brig` developing branch 75286a24

Apparently, I did not do syncing fully. The following steps lead to endless restore:

Use sync.conflict_strategy: marker (default)

1. Add a file on repo1: `echo dummy | brig stage --stdin /testfile`
2. at repo2: `brig sync`
3. at repo1: `brig rm /testfile; brig commit`
4. at repo1: `brig sync`
```
Syncing with `repo2`...
Added:
  ▩ ← /testfile
```

notice reappearance of `testfile` at repo1. 

This bug is a show stopper from using `brig` as working tree syncer.

Note: technically we do not have time line syncing but we clearly need it as a natural choice and extension of the `marker` strategy.

**Comment by sahib** [comment]
True, that is currently broken. I will add a test and see what's causing this.

> ▩ 

What is this icon by the way? I just see a shaded rectangle... 

> Note: technically we do not have time line syncing but we clearly need it as a natural choice and extension of the marker strategy.

Can you elaborate on what you mean by that? Do you mean to take over whatever came last time-wise as conflict strategy?

**Comment by sahib** [comment]
This should be fixed hopefully as of 401b1c12de2fb9a8746908ddfe5d945d6c1f5b51 - a mapper case was missing.

**Comment by evgmik** [comment]
Confirming 401b1c1. The removed file is not resurrected anymore.

Regarding ▩ symbol. I have added it to indicate an empty/ghost slot. I.e. removed file
would show up on diff as 
```
$ ./brig rm /testfile2 
$ ./brig diff -l
Removed:
  /testfile2 ← ▩
```

Similarly, if we create a new file it would be reported as
```
$ echo dummy | ./brig stage --stdin /testfile4 
$ ./brig diff -l                                                                                                                        
Added:
  ▩ ← /testfile4
```

I did it because I was somewhat confused when I compared diffs between repos, and it not always were clear at wich repo the file was deleted or removed. I hope you find it useful.
